test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,399 @@
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        let closest = plot.entityNearest({
+          x: xScale.scale(0) + 1,
+          y: yScale.scale(0) + 1
+        });
+
+        assert.deepEqual(closest.datum, dataset.data()[0], "correct datum has been retrieved");
+
         svg.remove();
       });
 
-      describe("after the scale has changed", () => {
-        beforeEach(() => {
-          xScale.domain([0, 3]);
-          yScale.domain([0, 9]);
+      it("ignores off-plot data points when retrieving the nearest Entity", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        yScale.domain([0, 1.9]);
+
+        let closest = plot.entityNearest({
+          x: xScale.scale(1),
+          y: xScale.scale(2)
         });
+        assert.deepEqual(closest.datum, dataset.data()[1], "correct datum has been retrieved");
 
-        it("the circles re-rendered properly", () => {
-          let circles = circlePlot.selections();
-          circles.each(getCirclePlotVerifier());
-          assert.strictEqual(circlesInArea, 4, "four circles were found in the render area");
-          svg.remove();
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain range", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({
+          min: xScale.scale(1),
+          max: xScale.scale(1)
+        }, {
+          min: yScale.scale(1),
+          max: yScale.scale(1)
+        });
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("does not return Entities whose center lies outside the range", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({
+          min: xScale.scale(1.001),
+          max: xScale.scale(1.001)
+        }, {
+          min: yScale.scale(1.001),
+          max: yScale.scale(1.001)
         });
+
+        assert.lengthOf(entities, 0, "no Entities retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain bounds", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        let entities = plot.entitiesIn({
+          topLeft: {
+            x: xScale.scale(1),
+            y: yScale.scale(1)
+          },
+          bottomRight: {
+            x: xScale.scale(1),
+            y: yScale.scale(1)
+          }});
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities centered at a given Point", () => {
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(1), y: yScale.scale(1) });
```

Comment:
Separate on multiple lines for consistency with the above. Either that or amend above to match this